sdk/storage/azure-storage-blob/azure/storage/blob/_blob_service_client.py

```diff
@@ -715,9 +704,10 @@ def get_container_client(self, container):
                 :dedent: 8
                 :caption: Getting the container client to interact with a specific container.
         """
-        try:
-            container_name = container.name
-        except AttributeError:
+        if isinstance(container, ContainerProperties):
+            if container.name is not None:
```

Comment:
In the models PR I commented on us trying to make `container.name` non-optional. This is why :) You would not need the None check here (and neither would our users)

Reply:
*facepalm* we just slap it in AFTER the constructor. fun.